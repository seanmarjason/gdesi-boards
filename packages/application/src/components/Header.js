import * as React from 'react';
import Stack from '@mui/material/Stack';
import NotificationsRoundedIcon from '@mui/icons-material/NotificationsRounded';
import NavbarBreadcrumbs from '../app/boards/components/NavbarBreadcrumbs';
import MenuButton from './MenuButton';

import Search from '../app/boards/components/Search';

export default function Header({ navigation }) {
  return (
    <Stack
      direction="row"
      sx={{
        display: { xs: 'none', md: 'flex' },
        width: '100%',
        alignItems: { xs: 'flex-start', md: 'center' },
        justifyContent: 'space-between',
        maxWidth: { sm: '100%', md: '1700px' },
        pt: 1.5,
      }}
      spacing={2}
    >
      <NavbarBreadcrumbs nav={navigation}/>
      <Stack direction="row" sx={{ gap: 1 }}>
        <Search />
        <MenuButton showBadge aria-label="Open notifications">
          <NotificationsRoundedIcon />
        </MenuButton>
      </Stack>
    </Stack>
  );
}
