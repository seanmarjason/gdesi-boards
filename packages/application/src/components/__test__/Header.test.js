import { render } from '@testing-library/react'
import Header from '../Header';

it('Header renders correctly', () => {
  const { container } = render(<Header navigation={['Root', 'Child']} />)
  expect(container).toMatchSnapshot()
})
