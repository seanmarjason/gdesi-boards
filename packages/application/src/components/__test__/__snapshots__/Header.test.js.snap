// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    class="MuiStack-root css-1sd6chr-MuiStack-root"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
              href="../../root"
            >
              Root
            </a>
          </p>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="NavigateNextRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9.31 6.71c-.39.39-.39 1.02 0 1.41L13.19 12l-3.88 3.88c-.39.39-.39 1.02 0 1.41s1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41L10.72 6.7c-.38-.38-1.02-.38-1.41.01"
            />
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1d2i4tu-MuiTypography-root"
          >
            Child
          </p>
        </li>
      </ol>
    </nav>
    <div
      class="MuiStack-root css-1oxswqj-MuiStack-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
        id=":r0:"
        tabindex="0"
        type="button"
      >
        Sign out
      </button>
    </div>
  </div>
</div>
`;
